src/backend/distributed/shared_library_init.c

```diff
@@ -327,6 +361,18 @@ RegisterCitusConfigVariables(void)
 		NULL, NULL, NULL);
 
 	DefineCustomBoolVariable(
+		"citus.log_distributed_deadlock_detection",
+		gettext_noop("Log distributed deadlock detection related processing in "
+					 "the server log"),
```

Comment:
I think this is severely conflating two different things. User targeted and developer targeted logging. It's sure not very friendly to users atm, and those are going to be the main consumers of this.